[SPARC] Remove CCIfConsecutiveRegs for f128 returns

koachan opened the pull request:
It appears that using it will result in callers mistakenly thinking that complex f128 returns is done the sret-way, when it should be returned in registers.

rorth commented:
> > ```
> > Builtins-sparc-sunos :: divtc3_test.c
> > Builtins-sparc-sunos :: multc3_test.c
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > Even for the 2-stage build, the two FAILing tests die with SIGILL
> 
> From a cursory look this seems like a genuine case of ABI mismatch bug; the caller expects an sret return but the callee returns the value in the register. Will look into it, thanks!

Good: I've now switched the buildbot over to a current `clang-22` as build compiler, so the only remaining `FAIL`s are the ones above.

> For the rest... Over here it turns out with or without this patch only the SIGILL ones failed too, so this might as well isn't needed. If so then I'll close or rework this PR to fix the SIGILL tests for the time being at least (?)

Agreed: this matches my testing with  and without this patch.

koachan committed:
[SPARC] Remove CCIfConsecutiveRegs for f128 returns

It appears that using it will result in callers mistakenly thinking that
complex f128 returns is done the sret-way.

koachan commented:
Okay, reworked the PR, it should fix the remaining SIGILL issues in 2-stage builds.

efriedma-quic reviewed (approved):
LGTM
